src/backend/distributed/metadata/node_metadata.c

```diff
@@ -196,6 +200,7 @@ master_add_inactive_node(PG_FUNCTION_ARGS)
 
 	int nodeId = AddNodeMetadata(nodeNameString, nodePort, &nodeMetadata,
 								 &nodeAlreadyExists);
+	TransactionModifiedNodeMetadata = true;
```

Comment:
Wondering if we should do this in the trigger as well, but might be overkill since we don't expect users to do that.